CodeStarterCamp_Week3/main.swift

```diff
@@ -10,36 +10,51 @@ struct Person {
     mutating func orderCoffee(coffeeShop: CoffeeShop, coffee: Coffee) {
         if let price = coffeeShop.menu[coffee] {
             if money >= price {
+                coffeeShop.numberOfOrder += 1
                 money = money - price
                 print("커피 가격은 \(price)원이고 남은 돈 \(money)원 입니다.")
+                coffeeShop.revenue += price
+                yagombucks.makeCoffee(from: name, coffee: coffee)
             } else {
-                print("돈이 부족하여 커피를 구매할 수 없습니다.")
+                print("잔액이 \(price - money)원만큼 부족합니다.")
             }
-            
         }
     }
 }
 
 
 class CoffeeShop {
     var shopName: String
-    var revenue: Int?
+    var revenue: Int {
+        didSet{
+            print("가게 매출은 \(revenue)원 입니다")
+        }
+    }
     var barista: Person
     var menu: [Coffee: Int] = [:]
-    var pickUpTable: [String] = []
+    var pickUpTable: [String] = [] {
+        didSet{
+            if var newCoffee = pickUpTable.last {
+                print("\(newCoffee) (이/가) 픽업대로 올라갑니다.")
+            }
+        }
+    }
+    var numberOfOrder: Int = 0
     
-    func makeCoffee(ordererName: String, coffee: Coffee) {
-        pickUpTable.append("\(coffee)")
-        for num in 0...(pickUpTable.count - 1) {
-            print("\(ordererName)님 주문하신 \(pickUpTable[num]) 나왔습니다.")
-            pickUpTable.removeAll()
+    func makeCoffee(from name: String, coffee: Coffee) {
+        if numberOfOrder > 0 {
+            pickUpTable.append("\(coffee)")
+            for num in 0...(pickUpTable.count - 1) {
+                print("\(name)님 주문하신 \(pickUpTable[num])(이/가) 나왔습니다.")
+            }
```

Comment:
`subscript` 때문에 유지했다고 하셨는데 이렇게 하면 되지 않을까요? 👀

```suggestion
            for coffee in pickUpTable {
                print("\(name)님 주문하신 \(coffee)(이/가) 나왔습니다.")
            }
```

코드를 읽을 때 주문한 `pickUpTable[num]이 나왔다` 보다는 `coffee가 나왔다`라고 써있는게 훨씬 이해하기 편해 보이네요!